`Development`: Add database tests run configurations and docs (#7658)
ls1intum/Artemis: docs/dev/setup/local-database-tests.rst

.. _local_database_tests:

Running database tests locally
---------------------------------

Prerequisites
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Database tests are standard server integration tests executed against real database (MySQL or Postgres) started as Docker containers. Gradle test task will automatically pull necessary images and start Docker containers.

Reviewer: ```suggestion
Database tests are the standard server integration tests executed against a real database (MySQL or Postgres) started as Docker containers. The Gradle test task will automatically pull the necessary images and start the containers.
```

> Before running those tests make sure that the following steps were executed:

Reviewer 2: Missing "," after "those tests".
Please consider not using passive voice and therefore rephrasing the "Before running..." sentence.

> Before running those tests, make sure that the following steps were executed:

Reviewer: ```suggestion
Before running those tests, make sure you have got a working Docker setup:
```

> 1. Install `Docker Desktop <https://docs.docker.com/desktop/#docker-for-mac>`__ or `Docker Engine and Docker CLI <https://docs.docker.com/engine/install/>`__.

Reviewer: ```suggestion
1. Install `Docker Desktop <https://docs.docker.com/desktop/>`__ or `Docker Engine and Docker CLI <https://docs.docker.com/engine/install/>`__.
```
Non-mac-specific link

> 2. Enable `remote access for Docker daemon <https://docs.docker.com/config/daemon/remote-access/>`__.

Reviewer: ```suggestion
2. Enable `remote access for the Docker daemon <https://docs.docker.com/config/daemon/remote-access/>`__.
```

Executing tests from Intellij
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Database tests can be executed by running of the run configurations provided in the repository:

Reviewer 2: Please consider not using passive voice and therefore rephrasing the "Database tests..." sentence.
In the same sentence "of" is not needed. You can remove it.

* For MySQL tests: `Artemis Server Tests MySQL`.
* For Postgres tests: `Artemis Server Tests Postgres`.

> Or by creating a custom configuration. Screenshot below presents an example of such run configuration:

Reviewer 2: I would change "Or ..." --> to  "Or alternatively, ..."
Add an article to "Screenshot ..." --> "The screenshot ..."
Add "a" before "... such run ..." --> "... such a run ..."

.. figure:: intellij-postgres-tests-run-configuration.png

>    :align: center

Reviewer: ```suggestion
   :align: center
   :scale: 50 %
```
The screenshot is pretty big and filly nearly the complete screen without scaling.

>    :alt: An example of IntelliJ run configuration fro starting Postgres server tests.

Reviewer 3: Typo here in fro, should be from.

> Executing tests from console

Reviewer 2: Add an article to "... console" --> "... the console"

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Database tests can be executed by running the dedicated Gradle command:

Reviewer 2: Please consider not using passive voice and therefore rephrasing the "Database tests..." sentence.
